p4-16/spec/P4-16-spec.mdk

```diff
@@ -3604,14 +3570,14 @@ expressions of type `int` must be compile-time known values.  The type
 - Truncating integer division between positive values, denoted by `/`.
 - Modulo between positive values, denoted by `%`.
 - Arithmetic shift left and right denoted by `<<` and `>>`. These operations produce an `int` result.
-  The right operand must be constant and positive.
+  The right operand must be either an unsigned constant of type `bit<S>` or a non-negative integer literal.
```

Comment:
Can it be a mathematical (unlimited-width) integer constant?

Reply:
Good point. The shifts of bit<w> and int<w> uses the wording of "a non-negative integer literal", but indeed a integer constant is also "constant and positive".